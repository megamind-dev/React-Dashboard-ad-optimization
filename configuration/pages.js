import {
    AUTH_FORGOT_PASSWORD_PAGE,
    AUTH_PAGE,
    AUTH_SIGNUP_PAGE,
    AUTOMATION_PAGE,
    BRAND_CAMPAIGN_KEYWORD_PAGE,
    BRAND_CAMPAIGN_PAGE,
    BRAND_CAMPAIGN_PRODUCT_AD_PAGE,
    BRAND_PAGE,
    BRANDS_SUMMARY_PAGE,
    CAMPAIGN_KEYWORD_PAGE,
    CAMPAIGN_PAGE,
    CAMPAIGN_PRODUCT_AD_PAGE,
    CAMPAIGNS_SUMMARY_PAGE,
    HOME_PAGE,
    KEYWORD_PAGE,
    KEYWORDS_SUMMARY_PAGE,
    LABELS_SUMMARY_PAGE,
    PRODUCT_PAGE,
    PRODUCT_PRODUCT_AD_PAGE,
    PRODUCTS_SUMMARY_PAGE,
    PROFILE_PAGE,
    SOV_KEYWORD_PAGE,
    SOV_KEYWORD_SEARCH_RESULT_PAGE,
    SOV_KEYWORDS_SUMMARY_PAGE,
    ORGANIZATION_PAGE,
    ORGANIZATION_GROUP_PAGE,
} from 'constants/pages'

export const PAGES_AND_PATHS = [
    {
        page: BRANDS_SUMMARY_PAGE,
        path: '/brands',
    },
    {
        page: BRAND_PAGE,
        path: '/brands/:brandId',
    },
    {
        page: BRAND_CAMPAIGN_PAGE,
        path: '/brands/:brandId/campaigns/:campaignId',
    },
    {
        page: BRAND_CAMPAIGN_KEYWORD_PAGE,
        path: '/brands/:brandId/campaigns/:campaignId/keywords/:keywordId',
    },
    {
        page: BRAND_CAMPAIGN_PRODUCT_AD_PAGE,
        path: '/brands/:brandId/campaigns/:campaignId/product-ads/:productAdId',
    },
    {
        page: CAMPAIGNS_SUMMARY_PAGE,
        path: '/campaigns',
    },
    {
        page: CAMPAIGN_PAGE,
        path: '/campaigns/:campaignId',
    },
    {
        page: CAMPAIGN_KEYWORD_PAGE,
        path: '/campaigns/:campaignId/keywords/:keywordId',
    },
    {
        page: CAMPAIGN_PRODUCT_AD_PAGE,
        path: '/campaigns/:campaignId/product-ads/:productAdId',
    },
    {
        page: PRODUCTS_SUMMARY_PAGE,
        path: '/products',
    },
    {
        page: PRODUCT_PAGE,
        path: '/products/:asin/:countryCode',
    },
    {
        page: PRODUCT_PRODUCT_AD_PAGE,
        path: '/products/:asin/:countryCode/product-ads/:productAdId',
    },
    {
        page: KEYWORDS_SUMMARY_PAGE,
        path: '/keywords',
    },
    {
        page: KEYWORD_PAGE,
        path: '/keywords/:keywordId',
    },
    {
        page: LABELS_SUMMARY_PAGE,
        path: '/labels',
    },
    {
        page: SOV_KEYWORDS_SUMMARY_PAGE,
        path: '/sov',
    },
    {
        page: SOV_KEYWORD_PAGE,
        path: '/sov/:sovKeywordId',
    },
    {
        page: SOV_KEYWORD_SEARCH_RESULT_PAGE,
        path: '/sov/:sovKeywordId/search/:scheduledDate',
    },
    {
        page: PROFILE_PAGE,
        path: '/profile',
    },
    {
        page: AUTH_PAGE,
        path: '/auth',
    },
    {
        page: AUTH_SIGNUP_PAGE,
        path: '/auth/signup',
    },
    {
        page: AUTH_FORGOT_PASSWORD_PAGE,
        path: '/auth/forgot',
    },
    {
        page: ORGANIZATION_PAGE,
        path: '/profile/organizations/:organizationId',
    },
    {
        page: ORGANIZATION_GROUP_PAGE,
        path:
            '/profile/organizations/:organizationId/groups/:organizationGroupId',
    },
    {
        page: AUTOMATION_PAGE,
        path: '/automation',
    },
    {
        page: HOME_PAGE,
        path: '/',
    },
]
