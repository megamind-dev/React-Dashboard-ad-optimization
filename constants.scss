// colors
$primary-color: #01c7d1;
$body-bg-color: #eff2f4;
$gray-background-color: #cccccc;
$white-background-color: #ffffff;
$card-border-color: #e8e8e8;
$danger-color: #f41e28;
$success-color: #52c41a;
$warning-color: #faad14;
$ghost-white: rgba(255, 255, 255, 0.2);
$text-primary: rgba(0, 0, 0, 0.85);
$text-muted: rgba(0, 0, 0, 0.65);
$text-light: rgba(0, 0, 0, 0.5);
$text-extra-light: rgba(0, 0, 0, 0.3);
$popover-mask-color: rgba(255, 255, 255, 0.5);
$side-menu-item-color: rgba(255, 255, 255, 0.5);
$action-button-color-light: #9cc681;
$action-button-color-medium: #8ab46f;
$action-button-color-dark: #68914d;

// dimensions
$row-padding: 15px;
$card-padding: 16px 24px;
$filter-panel-padding: 16px 24px;
$btn-right-margin: 10px;
$content-horiz-padding: 24px;
$icon-horizontal-margin: 5px;

// icon sizes
$medium-icon-size: 25px;
$small-icon-size: 20px;
$extra-small-icon-size: 12px;

// font sizes
$font-size-xxlarge: 28px;
$font-size-xlarge: 20px;
$font-size-large: 18px;
$font-size-medium: 14px;
$font-size-smaller: 12.6px;
$font-size-smallest: 11px;

// font weights
$font-weight-light: 300;
$font-weight-normal: 400;
$font-weight-semi-bold: 500;
$font-weight-bold: 700;

@mixin box-shadow {
  -webkit-box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16);
  -moz-box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16);
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16);
}
