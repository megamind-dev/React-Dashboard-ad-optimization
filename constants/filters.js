export const AGGREGATION = 'aggregation'

export const BRANDS = 'brands'

export const CAMPAIGN_BUDGET = 'campaignBudget'
export const CAMPAIGN_DAYPARTINGS = 'campaignDaypartings'
export const CAMPAIGN_NAME = 'campaignName'
export const CAMPAIGN_STATES = 'campaignStates'
export const CAMPAIGN_TARGETING_TYPES = 'campaignTargetingTypes'
export const CAMPAIGNS = 'campaigns'

export const COUNTRIES = 'countries'
export const REGIONS = 'regions'

export const DATES = 'dates'

export const FACT_TYPES = 'factTypes'

export const KEYWORD_MATCH_TYPES = 'keywordMatchTypes'
export const KEYWORDS = 'keywords'

export const PRODUCT_ASINS = 'productAsins'
export const PRODUCT_TITLES = 'productTitles'

export const LABELS = 'labels'

export const SOV_AGGREGATION = 'sovAggregation'
export const SOV_COUNTRIES = 'sovCountries'
export const SOV_BRANDS = 'sovBrands'
export const SOV_FOLDS = 'sovFolds'
export const SOV_KEYWORDS = 'sovKeywords'
export const SOV_KEYWORD_CATEGORIES = 'sovKeywordCategories'
export const SOV_LANGUAGES = 'sovLanguages'
export const SOV_RESULT_TYPES = 'sovResultTypes'
export const SOV_SEARCH_TIMES = 'searchTimes'
export const SOV_STATES = 'sovStates'

export const GREATER_THAN = 'gt'
export const EQUALS = 'exact'
export const LESS_THAN = 'lt'
export const GREATER_THAN_OR_EQUAL = 'gte'
export const LESS_THAN_OR_EQUAL = 'lte'

export const NUMBER_RANGE_OPERATOR_OPTIONS = [
    { label: 'Greater Than', value: GREATER_THAN },
    { label: 'Equals', value: EQUALS },
    { label: 'Less Than', value: LESS_THAN },
    { label: 'Greater Than or Equal To', value: GREATER_THAN_OR_EQUAL },
    { label: 'Less Than or Equal To', value: LESS_THAN_OR_EQUAL },
]

export const NUMBER_RANGE_OPERATOR_DISPLAY = {
    [GREATER_THAN]: '>',
    [EQUALS]: '=',
    [LESS_THAN]: '<',
    [GREATER_THAN_OR_EQUAL]: '>=',
    [LESS_THAN_OR_EQUAL]: '<=',
}
