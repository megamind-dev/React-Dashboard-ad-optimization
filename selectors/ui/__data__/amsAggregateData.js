export const amsAggregateData = {
    count: 4,
    next: null,
    previous: null,
    results: [
        {
            attributed_sales_14_day__sum: 438629.22,
            ctr: 0.56,
            profile_id: '320788626716217',
            profile: {
                id: '320788626716217',
                brand_name: 'Acme Brand 1',
                region: 'NA',
                brand_entity_id: 'ENTITY1TOLU9YL3AZ8G',
                marketplace_string_id: 'ATVPDKIKX0DER',
                country_code: 'US',
                timezone: 'America/Los_Angeles',
                created_date: '2018-11-07T23:28:36.462251Z',
                updated_date: '2018-11-07T23:28:36.630581Z',
                currency_code: 'USD',
            },
        },
        {
            attributed_sales_14_day__sum: 227308,
            ctr: 0.65,
            profile_id: '2267480161047622',
            profile: {
                id: '2267480161047622',
                brand_name: 'Acme Brand 2',
                region: 'NA',
                brand_entity_id: 'ENTITY3OOEMY9H6L0MZ',
                marketplace_string_id: 'ATVPDKIKX0DER',
                country_code: 'US',
                timezone: 'America/Los_Angeles',
                created_date: '2018-11-07T23:28:36.462251Z',
                updated_date: '2018-11-07T23:28:36.630581Z',
                currency_code: 'USD',
            },
        },
        {
            attributed_sales_14_day__sum: 200364.97,
            ctr: 0.11,
            profile_id: '356171147073308',
            profile: {
                id: '356171147073308',
                brand_name: 'Acme Brand 3',
                region: 'NA',
                brand_entity_id: 'ENTITY2Y99QB5Y0LQ7V',
                marketplace_string_id: 'ATVPDKIKX0DER',
                country_code: 'US',
                timezone: 'America/Los_Angeles',
                created_date: '2018-11-07T23:28:36.462251Z',
                updated_date: '2018-11-07T23:28:36.630581Z',
                currency_code: 'USD',
            },
        },
        {
            attributed_sales_14_day__sum: 174992.39,
            ctr: 0.15,
            profile_id: '3931123856677272',
            profile: {
                id: '3931123856677272',
                brand_name: 'Acme Brand 4',
                region: 'NA',
                brand_entity_id: 'ENTITY33WK215LJ5HLQ',
                marketplace_string_id: 'ATVPDKIKX0DER',
                country_code: 'US',
                timezone: 'America/Los_Angeles',
                created_date: '2018-11-07T23:28:36.462251Z',
                updated_date: '2018-11-07T23:28:36.630581Z',
                currency_code: 'USD',
            },
        },
    ],
}
