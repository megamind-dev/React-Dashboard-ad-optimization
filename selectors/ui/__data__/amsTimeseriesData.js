export const headlineSearchTimeseriesData = [
    {
        clicks__sum: 3972,
        cost__sum: 2476.06,
        impressions__sum: 357965,
        attributed_conversions_14_day__sum: 354,
        attributed_conversions_14_day_same_sku__sum: 34,
        attributed_sales_14_day__sum: 34161.48,
        attributed_sales_14_day_same_sku__sum: 3671.21,
        ctr: 0.0110960569,
        roas: 13.7967314677,
        spc: 8.6005733095,
        cpc: 0.623377597,
        cpa: 6.9945079533,
        cpm: 6.9170332728,
        spm: 95.4324506174,
        aov: 96.5013479809,
        conversion_rate_clicks: 0.0891238671,
        conversion_rate_impressions: 0.0009889235,
        report_date: '2018-11-23T00:00:00Z',
    },
    {
        clicks__sum: 2568,
        cost__sum: 1442.67,
        impressions__sum: 237866,
        attributed_conversions_14_day__sum: 118,
        attributed_conversions_14_day_same_sku__sum: 9,
        attributed_sales_14_day__sum: 11560.42,
        attributed_sales_14_day_same_sku__sum: 352.96,
        ctr: 0.0107959944,
        roas: 8.0132291232,
        spc: 4.5017207398,
        cpc: 0.5617861003,
        cpa: 12.2259890298,
        cpm: 6.0650395833,
        spm: 48.600551822,
        aov: 97.9696513534,
        conversion_rate_clicks: 0.0459501558,
        conversion_rate_impressions: 0.0004960776,
        report_date: '2018-11-22T00:00:00Z',
    },
    {
        clicks__sum: 2420,
        cost__sum: 1290.41,
        impressions__sum: 218753,
        attributed_conversions_14_day__sum: 120,
        attributed_conversions_14_day_same_sku__sum: 7,
        attributed_sales_14_day__sum: 8931.55,
        attributed_sales_14_day_same_sku__sum: 825.17,
        ctr: 0.0110627054,
        roas: 6.9214766377,
        spc: 3.6907236625,
        cpc: 0.5332277859,
        cpa: 10.753427015,
        cpm: 5.8989419199,
        spm: 40.8293886856,
        aov: 74.4295938595,
        conversion_rate_clicks: 0.0495867769,
        conversion_rate_impressions: 0.0005485639,
        report_date: '2018-11-21T00:00:00Z',
    },
    {
        clicks__sum: 2480,
        cost__sum: 1316.3,
        impressions__sum: 221219,
        attributed_conversions_14_day__sum: 118,
        attributed_conversions_14_day_same_sku__sum: 9,
        attributed_sales_14_day__sum: 7792.61,
        attributed_sales_14_day_same_sku__sum: 734.81,
        ctr: 0.0112106103,
        roas: 5.9200976732,
        spc: 3.1421821048,
        cpc: 0.5307652472,
        cpa: 11.1550662119,
        cpm: 5.950202347,
        spm: 35.2257790694,
        aov: 66.0390815251,
        conversion_rate_clicks: 0.0475806452,
        conversion_rate_impressions: 0.0005334081,
        report_date: '2018-11-20T00:00:00Z',
    },
]

export const sponsoredProductTimeseriesData = [
    {
        clicks__sum: 21651,
        cost__sum: 13459.66,
        impressions__sum: 5967377,
        attributed_conversions_14_day__sum: 1470,
        attributed_conversions_14_day_same_sku__sum: 1061,
        attributed_sales_14_day__sum: 84520.82,
        attributed_sales_14_day_same_sku__sum: 60384.11,
        ctr: 0.0036282273,
        roas: 6.2795636744,
        spc: 3.9037834419,
        cpc: 0.6216647596,
        cpa: 9.1562338161,
        cpm: 2.2555410375,
        spm: 14.163813565,
        aov: 57.4971532655,
        conversion_rate_clicks: 0.0678952473,
        conversion_rate_impressions: 0.0002463394,
        attributed_conversions_1_day__sum: 1470,
        attributed_conversions_1_day_same_sku__sum: 1061,
        attributed_conversions_7_day__sum: 1470,
        attributed_conversions_7_day_same_sku__sum: 1061,
        attributed_conversions_30_day__sum: 1470,
        attributed_conversions_30_day_same_sku__sum: 1061,
        attributed_sales_1_day__sum: 84520.82,
        attributed_sales_1_day_same_sku__sum: 60384.11,
        attributed_sales_7_day__sum: 84520.82,
        attributed_sales_7_day_same_sku__sum: 60384.11,
        attributed_sales_30_day__sum: 84520.82,
        attributed_sales_30_day_same_sku__sum: 60384.11,
        attributed_units_ordered_1_day__sum: 1555,
        attributed_units_ordered_7_day__sum: 1555,
        attributed_units_ordered_14_day__sum: 1555,
        attributed_units_ordered_30_day__sum: 1555,
        report_date: '2018-11-23T00:00:00Z',
    },
    {
        clicks__sum: 14030,
        cost__sum: 7582.7,
        impressions__sum: 3247452,
        attributed_conversions_14_day__sum: 640,
        attributed_conversions_14_day_same_sku__sum: 441,
        attributed_sales_14_day__sum: 41116.48,
        attributed_sales_14_day_same_sku__sum: 27991.86,
        ctr: 0.0043203102,
        roas: 5.4224067915,
        spc: 2.9306113136,
        cpc: 0.5404631977,
        cpa: 11.8479666615,
        cpm: 2.3349686657,
        spm: 12.6611499508,
        aov: 64.2444948908,
        conversion_rate_clicks: 0.045616536,
        conversion_rate_impressions: 0.0001970776,
        attributed_conversions_1_day__sum: 633,
        attributed_conversions_1_day_same_sku__sum: 436,
        attributed_conversions_7_day__sum: 640,
        attributed_conversions_7_day_same_sku__sum: 441,
        attributed_conversions_30_day__sum: 640,
        attributed_conversions_30_day_same_sku__sum: 441,
        attributed_sales_1_day__sum: 40395.61,
        attributed_sales_1_day_same_sku__sum: 27352.02,
        attributed_sales_7_day__sum: 41116.48,
        attributed_sales_7_day_same_sku__sum: 27991.86,
        attributed_sales_30_day__sum: 41116.48,
        attributed_sales_30_day_same_sku__sum: 27991.86,
        attributed_units_ordered_1_day__sum: 671,
        attributed_units_ordered_7_day__sum: 678,
        attributed_units_ordered_14_day__sum: 678,
        attributed_units_ordered_30_day__sum: 678,
        report_date: '2018-11-22T00:00:00Z',
    },
    {
        clicks__sum: 11823,
        cost__sum: 5720.78,
        impressions__sum: 2362859,
        attributed_conversions_14_day__sum: 626,
        attributed_conversions_14_day_same_sku__sum: 462,
        attributed_sales_14_day__sum: 35794.39,
        attributed_sales_14_day_same_sku__sum: 26679.51,
        ctr: 0.0050036841,
        roas: 6.2569024986,
        spc: 3.0275213929,
        cpc: 0.483869038,
        cpa: 9.138632007,
        cpm: 2.4211278102,
        spm: 15.1487606449,
        aov: 57.1795294386,
        conversion_rate_clicks: 0.0529476444,
        conversion_rate_impressions: 0.0002649333,
        attributed_conversions_1_day__sum: 604,
        attributed_conversions_1_day_same_sku__sum: 447,
        attributed_conversions_7_day__sum: 626,
        attributed_conversions_7_day_same_sku__sum: 462,
        attributed_conversions_30_day__sum: 626,
        attributed_conversions_30_day_same_sku__sum: 462,
        attributed_sales_1_day__sum: 33185.44,
        attributed_sales_1_day_same_sku__sum: 24906,
        attributed_sales_7_day__sum: 35794.39,
        attributed_sales_7_day_same_sku__sum: 26679.51,
        attributed_sales_30_day__sum: 35794.39,
        attributed_sales_30_day_same_sku__sum: 26679.51,
        attributed_units_ordered_1_day__sum: 668,
        attributed_units_ordered_7_day__sum: 690,
        attributed_units_ordered_14_day__sum: 690,
        attributed_units_ordered_30_day__sum: 690,
        report_date: '2018-11-21T00:00:00Z',
    },
    {
        clicks__sum: 12018,
        cost__sum: 5839.85,
        impressions__sum: 2397061,
        attributed_conversions_14_day__sum: 659,
        attributed_conversions_14_day_same_sku__sum: 453,
        attributed_sales_14_day__sum: 36336.42,
        attributed_sales_14_day_same_sku__sum: 25567.49,
        ctr: 0.0050136396,
        roas: 6.2221520795,
        spc: 3.0234994593,
        cpc: 0.4859250338,
        cpa: 8.8616799024,
        cpm: 2.4362530014,
        spm: 15.1587366786,
        aov: 55.1387200329,
        conversion_rate_clicks: 0.054834415,
        conversion_rate_impressions: 0.00027492,
        attributed_conversions_1_day__sum: 593,
        attributed_conversions_1_day_same_sku__sum: 412,
        attributed_conversions_7_day__sum: 659,
        attributed_conversions_7_day_same_sku__sum: 453,
        attributed_conversions_30_day__sum: 659,
        attributed_conversions_30_day_same_sku__sum: 453,
        attributed_sales_1_day__sum: 31176.18,
        attributed_sales_1_day_same_sku__sum: 21943.13,
        attributed_sales_7_day__sum: 36336.42,
        attributed_sales_7_day_same_sku__sum: 25567.49,
        attributed_sales_30_day__sum: 36336.42,
        attributed_sales_30_day_same_sku__sum: 25567.49,
        attributed_units_ordered_1_day__sum: 653,
        attributed_units_ordered_7_day__sum: 719,
        attributed_units_ordered_14_day__sum: 719,
        attributed_units_ordered_30_day__sum: 719,
        report_date: '2018-11-20T00:00:00Z',
    },
]
