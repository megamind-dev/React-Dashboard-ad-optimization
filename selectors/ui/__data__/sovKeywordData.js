export const sovKeywordData = {
    count: 111,
    next: null,
    previous: null,
    results: [
        {
            search_time_day: '2018-11-21T00:00:00Z',
            metadata__brand: 'Amida',
            count: 25,
            pixels__sum: 6005575,
        },
        {
            search_time_day: '2018-11-21T00:00:00Z',
            metadata__brand: 'HP',
            count: 148,
            pixels__sum: 24841325,
        },
        {
            search_time_day: '2018-11-21T00:00:00Z',
            metadata__brand: 'Inkfirst',
            count: 175,
            pixels__sum: 43585525,
        },
        {
            search_time_day: '2018-11-21T00:00:00Z',
            metadata__brand: null,
            count: 24,
            pixels__sum: 5216400,
        },
        {
            search_time_day: '2018-11-20T00:00:00Z',
            metadata__brand: 'Amida',
            count: 25,
            pixels__sum: 6005575,
        },
        {
            search_time_day: '2018-11-20T00:00:00Z',
            metadata__brand: 'HP',
            count: 150,
            pixels__sum: 24876525,
        },
        {
            search_time_day: '2018-11-20T00:00:00Z',
            metadata__brand: 'Inkfirst',
            count: 175,
            pixels__sum: 43585525,
        },
        {
            search_time_day: '2018-11-20T00:00:00Z',
            metadata__brand: 'Ink & Toner 4 You',
            count: 25,
            pixels__sum: 6005575,
        },
        {
            search_time_day: '2018-11-20T00:00:00Z',
            metadata__brand: 'LD',
            count: 25,
            pixels__sum: 6005575,
        },
        {
            search_time_day: '2018-11-20T00:00:00Z',
            metadata__brand: null,
            count: 25,
            pixels__sum: 5433750,
        },
        {
            search_time_day: '2018-11-19T00:00:00Z',
            metadata__brand: 'Amida',
            count: 24,
            pixels__sum: 5765352,
        },
        {
            search_time_day: '2018-11-19T00:00:00Z',
            metadata__brand: 'HP',
            count: 142,
            pixels__sum: 23846264,
        },
        {
            search_time_day: '2018-11-19T00:00:00Z',
            metadata__brand: 'Inkfirst',
            count: 168,
            pixels__sum: 41842104,
        },
        {
            search_time_day: '2018-11-19T00:00:00Z',
            metadata__brand: 'Ink & Toner 4 You',
            count: 24,
            pixels__sum: 5765352,
        },
        {
            search_time_day: '2018-11-19T00:00:00Z',
            metadata__brand: 'LD',
            count: 24,
            pixels__sum: 5765352,
        },
        {
            search_time_day: '2018-11-19T00:00:00Z',
            metadata__brand: 'Moustache',
            count: 24,
            pixels__sum: 5765352,
        },
        {
            search_time_day: '2018-11-19T00:00:00Z',
            metadata__brand: 'PayForLess',
            count: 24,
            pixels__sum: 5765352,
        },
        {
            search_time_day: '2018-11-19T00:00:00Z',
            metadata__brand: 'SaveOnMany',
            count: 48,
            pixels__sum: 11530704,
        },
    ],
}
