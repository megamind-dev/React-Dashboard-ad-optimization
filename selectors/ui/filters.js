import createCachedSelector from 're-reselect'
import get from 'lodash/get'
import pickBy from 'lodash/pickBy'

import { FILTERS, FILTER_SETTINGS } from 'constants/reducerKeys'
import { AGGREGATION } from 'constants/filters'

import { selectDomainState } from './ui'

export const selectPageFilters = createCachedSelector(
    selectDomainState,
    (state, pageName) => pageName,
    (uiState, pageName) => get(uiState, [pageName, FILTERS], {})
)((state, pageName) => pageName)

export const selectPageFilterSettings = createCachedSelector(
    selectDomainState,
    (state, pageName) => pageName,
    (uiState, pageName) => get(uiState, [pageName, FILTER_SETTINGS], {})
)((state, pageName) => pageName)

export const selectVisiblePageFilters = createCachedSelector(
    (state, pageName) => selectPageFilters(state, pageName),
    (state, pageName) => selectPageFilterSettings(state, pageName),
    (filters, { anchored = [], displayState }) =>
        pickBy(
            filters,
            (value, key) => anchored.includes(key) || displayState[key]
        )
)((state, pageName) => pageName)

export const selectPageAggregation = createCachedSelector(
    selectDomainState,
    (state, pageName) => pageName,
    (domainState, pageName) =>
        get(domainState, [pageName, FILTERS, AGGREGATION])
)((state, pageName) => pageName)
