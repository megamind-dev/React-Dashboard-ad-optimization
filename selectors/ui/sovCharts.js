import createCachedSelector from 're-reselect'
import get from 'lodash/get'
import groupBy from 'lodash/groupBy'
import last from 'lodash/last'
import isEmpty from 'lodash/isEmpty'
import flow from 'lodash/flow'
import map from 'lodash/fp/map'
import orderBy from 'lodash/fp/orderBy'

import moment from 'utilities/moment'
import { SOV_CHART } from 'constants/reducerKeys'
import { SOV_AGGREGATION, SOV_BRANDS } from 'constants/filters'
import { UNDEFINED_BRAND } from 'constants/formatting'

import { selectDomainValue } from './ui'
import { selectPageFilters } from './filters'

const mapWithKey = map.convert({ cap: false })

const selectPageSovChartData = (state, pageName) =>
    selectDomainValue(state, [pageName, SOV_CHART, 'data'])

export const selectPageSovChartWeight = (state, pageName) =>
    selectDomainValue(state, [pageName, SOV_CHART, 'weight'])

export const selectDataForSovChart = createCachedSelector(
    selectPageSovChartData,
    selectPageSovChartWeight,
    selectPageFilters,
    (chartData, weightKey, filters) => {
        // format data as a highcharts series
        const aggregate = filters[SOV_AGGREGATION]
        const filteredBrands = get(filters, SOV_BRANDS, [])
        const lowerCaseFilteredBrandNames = filteredBrands.map(brandFilter =>
            brandFilter.value.toLowerCase()
        )

        // group data by brands
        const brandGroups = groupBy(
            chartData,
            result => result.metadata__brand || UNDEFINED_BRAND
        )

        // used to calculate group with most data points
        const maxDataPoints = Math.max(
            ...Object.values(brandGroups).map(brandData => brandData.length)
        )

        const series = flow(
            mapWithKey((brandData, brandName) => {
                const selected = lowerCaseFilteredBrandNames.includes(
                    brandName.toLowerCase()
                )
                const points = flow(
                    map(result => ({
                        x: moment(result[aggregate]).valueOf(),
                        y: result[weightKey],
                    })),

                    // Sort in ascending X order
                    orderBy(point => point.x, 'asc')
                )(brandData)

                return {
                    name: brandName,
                    data: points,
                    selected,

                    // If brands filter is not empty, show areas in transparency by default
                    fillOpacity: isEmpty(filteredBrands) ? null : 0.3,

                    // Style for brands selected in filter
                    ...(selected && {
                        fillOpacity: 1, // Fill area with solid color
                        lineColor: '#ffffff', // Highlight area line with white color
                        marker: {
                            symbol: 'diamond', // Show marker in diamond
                            radius: 8, // Enlarge marker
                        },
                    }),
                }
            }),

            orderBy(
                [
                    // Place brands selected in filter on top
                    brandSeries => (brandSeries.selected ? 1 : 0),

                    // Sort in desc order of y-axis, using most recent data point
                    brandSeries => last(brandSeries.data).y,
                ],
                ['desc', 'desc']
            )
        )(brandGroups)

        return { series, maxDataPoints }
    }
)((state, pageName) => pageName)

export const selectLoadingForSovChart = (state, pageName) =>
    selectDomainValue(state, [pageName, SOV_CHART, 'loading'])
