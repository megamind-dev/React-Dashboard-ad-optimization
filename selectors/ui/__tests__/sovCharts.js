import { SOV_KEYWORD_PAGE } from 'constants/pages'
import { SOV_AGGREGATION_UNIT } from 'constants/reducerKeys'
import { AGGREGATION } from 'constants/filters'
import {
    fetchSovKeywordPageSovChartSuccess,
    updatePageFilter,
} from 'actions/ui'
import reducer from 'reducers'

import { selectDataForSovChart } from '../sovCharts'
import { sovKeywordData } from '../__data__/sovKeywordData'

describe('[Selectors] ui/sovCharts', () => {
    let state

    describe('selectDataForSovChart', () => {
        const pageName = SOV_KEYWORD_PAGE

        beforeEach(() => {
            // action to update sov data
            state = reducer(
                state,
                fetchSovKeywordPageSovChartSuccess(sovKeywordData)
            )
        })

        it('selects chart data correctly', () => {
            const { series } = selectDataForSovChart(state, pageName)
            expect(series).toMatchSnapshot()
        })

        it('memoizes chart settings', () => {
            // call the selector
            selectDataForSovChart(state, pageName)

            // get the actual selector that was used
            const selector = selectDataForSovChart.getMatchingSelector(
                state,
                pageName
            )

            // Reset computations to 0
            selector.resetRecomputations()

            // Call selector without state change and expect no new computations
            selectDataForSovChart(state, pageName)
            selectDataForSovChart(state, pageName)
            expect(selector.recomputations()).toEqual(0)

            // Update state with new aggregation
            state = reducer(
                state,
                updatePageFilter({
                    pageName,
                    data: {
                        key: AGGREGATION,
                        value: SOV_AGGREGATION_UNIT.MONTH,
                    },
                })
            )

            // Call selector after state change and expect a new computation
            selectDataForSovChart(state, pageName)
            selectDataForSovChart(state, pageName)
            expect(selector.recomputations()).toEqual(1)
        })
    })
})
