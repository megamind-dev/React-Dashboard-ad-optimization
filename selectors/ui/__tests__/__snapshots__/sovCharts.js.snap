// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Selectors] ui/sovCharts selectDataForSovChart selects chart data correctly 1`] = `
Array [
  Object {
    "data": Array [
      Object {
        "x": 1542585600000,
        "y": 168,
      },
      Object {
        "x": 1542672000000,
        "y": 175,
      },
      Object {
        "x": 1542758400000,
        "y": 175,
      },
    ],
    "fillOpacity": null,
    "name": "Inkfirst",
    "selected": false,
  },
  Object {
    "data": Array [
      Object {
        "x": 1542585600000,
        "y": 142,
      },
      Object {
        "x": 1542672000000,
        "y": 150,
      },
      Object {
        "x": 1542758400000,
        "y": 148,
      },
    ],
    "fillOpacity": null,
    "name": "HP",
    "selected": false,
  },
  Object {
    "data": Array [
      Object {
        "x": 1542585600000,
        "y": 48,
      },
    ],
    "fillOpacity": null,
    "name": "SaveOnMany",
    "selected": false,
  },
  Object {
    "data": Array [
      Object {
        "x": 1542585600000,
        "y": 24,
      },
      Object {
        "x": 1542672000000,
        "y": 25,
      },
      Object {
        "x": 1542758400000,
        "y": 25,
      },
    ],
    "fillOpacity": null,
    "name": "Amida",
    "selected": false,
  },
  Object {
    "data": Array [
      Object {
        "x": 1542585600000,
        "y": 24,
      },
      Object {
        "x": 1542672000000,
        "y": 25,
      },
    ],
    "fillOpacity": null,
    "name": "Ink & Toner 4 You",
    "selected": false,
  },
  Object {
    "data": Array [
      Object {
        "x": 1542585600000,
        "y": 24,
      },
      Object {
        "x": 1542672000000,
        "y": 25,
      },
    ],
    "fillOpacity": null,
    "name": "LD",
    "selected": false,
  },
  Object {
    "data": Array [
      Object {
        "x": 1542672000000,
        "y": 25,
      },
      Object {
        "x": 1542758400000,
        "y": 24,
      },
    ],
    "fillOpacity": null,
    "name": "Unknown Brand",
    "selected": false,
  },
  Object {
    "data": Array [
      Object {
        "x": 1542585600000,
        "y": 24,
      },
    ],
    "fillOpacity": null,
    "name": "Moustache",
    "selected": false,
  },
  Object {
    "data": Array [
      Object {
        "x": 1542585600000,
        "y": 24,
      },
    ],
    "fillOpacity": null,
    "name": "PayForLess",
    "selected": false,
  },
]
`;
